File: external_apps/mobile_android/app/src/main/java/com/example/mobile_android/data/analytics/CrashReporter.kt
package com.example.mobile_android.data.analytics

import android.content.Context
import android.util.Log
import dagger.hilt.android.qualifiers.ApplicationContext
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Crash Reporter
 * 
 * Note: Integrate with Firebase Crashlytics
 * 
 * Add to build.gradle:
 * implementation("com.google.firebase:firebase-crashlytics-ktx")
 */

@Singleton
class CrashReporter @Inject constructor(
    @ApplicationContext private val context: Context
) {
    
    /**
     * Log non-fatal exception
     */
    fun logException(throwable: Throwable, message: String? = null) {
        // TODO: Send to Crashlytics
        // FirebaseCrashlytics.getInstance().recordException(throwable)
        
        Log.e("CrashReporter", message ?: "Exception occurred", throwable)
    }
    
    /**
     * Log custom message
     */
    fun log(message: String) {
        // TODO: Add to Crashlytics log
        // FirebaseCrashlytics.getInstance().log(message)
        
        Log.d("CrashReporter", message)
    }
    
    /**
     * Set custom key
     */
    fun setCustomKey(key: String, value: String) {
        // TODO: Set Crashlytics custom key
        // FirebaseCrashlytics.getInstance().setCustomKey(key, value)
        
        Log.d("CrashReporter", "Custom Key: $key = $value")
    }
    
    /**
     * Set user identifier
     */
    fun setUserId(userId: String) {
        // TODO: Set Crashlytics user ID
        // FirebaseCrashlytics.getInstance().setUserId(userId)
        
        Log.d("CrashReporter", "User ID: $userId")
    }
    
    /**
     * Force crash (for testing)
     */
    fun forceCrash() {
        throw RuntimeException("Test crash")
    }
    
    /**
     * Check if crash reporting is enabled
     */
    fun isCrashReportingEnabled(): Boolean {
        // TODO: Check Crashlytics status
        // return FirebaseCrashlytics.getInstance().isCrashlyticsCollectionEnabled
        return true
    }
    
    /**
     * Enable/disable crash reporting
     */
    fun setCrashReportingEnabled(enabled: Boolean) {
        // TODO: Set Crashlytics collection
        // FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(enabled)
        
        Log.d("CrashReporter", "Crash reporting: $enabled")
    }
}

File: external_apps/mobile_ios/mobile_ios/Core/Analytics/CrashReporter.swift
//
//  CrashReporter.swift
//  mobile_ios
//
//  Created for Analytics & Monitoring
//

import Foundation

class CrashReporter {
    static let shared = CrashReporter()
    
    private init() {}
    
    /// Log non-fatal exception
    func logException(_ error: Error, message: String? = nil) {
        // TODO: Send to Crashlytics
        // Crashlytics.crashlytics().record(error: error)
        
        print("CrashReporter: \(message ?? "Exception occurred") - \(error.localizedDescription)")
    }
    
    /// Log custom message
    func log(_ message: String) {
        // TODO: Add to Crashlytics log
        // Crashlytics.crashlytics().log(message)
        
        print("CrashReporter: \(message)")
    }
    
    /// Set custom key
    func setCustomKey(_ key: String, value: String) {
        // TODO: Set Crashlytics custom key
        // Crashlytics.crashlytics().setCustomValue(value, forKey: key)
        
        print("Custom Key: \(key) = \(value)")
    }
    
    /// Set user identifier
    func setUserId(_ userId: String) {
        // TODO: Set Crashlytics user ID
        // Crashlytics.crashlytics().setUserID(userId)
        
        print("User ID: \(userId)")
    }
    
    /// Force crash (for testing)
    func forceCrash() {
        fatalError("Test crash")
    }
    
    /// Check if crash reporting is enabled
    func isCrashReportingEnabled() -> Bool {
        // TODO: Check Crashlytics status
        // return Crashlytics.crashlytics().isCrashlyticsCollectionEnabled()
        return true
    }
    
    /// Enable/disable crash reporting
    func setCrashReportingEnabled(_ enabled: Bool) {
        // TODO: Set Crashlytics collection
        // Crashlytics.crashlytics().setCrashlyticsCollectionEnabled(enabled)
        
        print("Crash reporting: \(enabled)")
    }
}
